### ISSUE TITLE
Slow performance while navigating songs-sites etc.

### ISSUE THREAD

**Issue by Psy-Virus** [ampache/ampache#1047]
Hey ho,
I forgot to write something as I saw #983 the first time...

The thing is, on ~big catalogs, at least on mine, ca. 150k songs, the navigation from site to site within "songs", "albums" or "artists" went slow in the past time (arount 3 seconds loading time). I don't know if it was some commits before or after this [PR](#983).
But it feels now, that the complete database is loaded or queried while you go from site to site.

Is there anything that could be done to improve the performance? Maybe also a limiting of the results?
And I think I also saw another issue facing the same problem, but I can't find it. So please don't kill me if I created a dup.

If I only dreamed of a similar issue, did anyone of you noticed this too?

:-)

**Comment by Afterster** [comment]
I don't get what you means with `from site to site`.
I don't think #983 PR has something to do with it.

And I think yes it is a duplicate with #723, please confirm :smile:

**Comment by Psy-Virus** [comment]
Hey @Afterster,
within Ampache, go to songs and you get the song list, like on artists and albums. And if you have more songs then the list will show at once, you get pages/sites/views/what ever. See the pic :-)
![ampachelistpages](https://cloud.githubusercontent.com/assets/6222200/11877310/ce7e3184-a4ee-11e5-961c-937caf755fa3.png)

And if I navigate through this pages... (no really, what's the right word for it? View?) the loading times are extreme. It was faster in the past. But I don't know if it has gone worse any time after #983 or before. I didn't mean that this PR had anything to do with it, sorry for confusion. :smile:

So as I see it, it seems to be / or feels like, that the queried Database entries aren't limited and the whole catalog entries are queried every time you go a page forward or backwards.

So maybe the performance for any catalog width could be, that only the entries are queried that should be shown. But there's also the fact, that for every page the results must firstly be sorted and the there must be a detailed query what entries should be pulled.

Like: 
View 1: Database, sort the results by artist and show me only the first 50 entries.
View 2: Database, sort the results by artist and show me only the entries from 51 to 100.
...

But I'm sure it was much faster before. In the past, only the first loading of albums/song/artists was slow. but now its every page/view/site thing... (I really don't know what's the right word for it) :laughing: 

I hope this clears it up :grinning:

**Comment by Afterster** [comment]
This is likely because more features are enabled. I'm going to close it as it is not an identified issue but more about a discussion (yes I don't like to have too much general issue opened, as you said before :wink: ). Try to give a try disabling unnecessary features and see if it's better.

**Comment by madeye-matt** [comment]
@Psy-Virus 

See https://github.com/ampache/ampache/issues/1322

Looks like the problem may be related to how many times tracks have been played.  

Are you able to run the following command against your ampache database?
```
select count(*) from object_count;
```

**Comment by Psy-Virus** [comment]
Hey @madeye-matt,
thanx for the further investigation (incl #1322). But I think on my side it's more the amount of songs (around 150k) 'cause the command only gave me `1624` (?)
And sorry for the late response, btw...
